widget/editor.go

```diff
@@ -936,6 +1171,153 @@ func (e *Editor) NumLines() int {
 	return len(e.lines)
 }
 
+func (e *Editor) IsSelected() bool {
+	return e.selection.len() > 0
+}
+
+// Selection returns the start and end of the selection, as offsets into the
+// editor text. start can be > end.
+func (e *Editor) Selection() (start, end int) {
+	return e.selection.start.ofs, e.selection.end.ofs
+}
+
+// SetSelection sets the selection start and end, which are offsets into the
+// editor text.
+func (e *Editor) SetSelection(start, end int) {
```

Comment:
SetSelection is maybe better named MoveAbs?

Reply:
How so?  I assume that's short for "move absolute".  Move what?  I guess it would move the selection, sure, and it's "absolute" in that it's not relative to the window or the line layout or what-have-you.  But no, I think for what it does, SetSelection is the appropriate name.

Maybe this is predicated on making the selection be caret + offset?  (Which your earlier question "Is caret.pos ever != selection.start?" sort of alludes to.) 
 In which case setting the selection would have to (be prepared to) move the caret.  Even then I think I'd prefer SetSelection, especially given the other function Selection to get the selection offsets.

Reply:
Ok, let's leave the exported names for now.

> In which case setting the selection would have to (be prepared to) move the caret. Even then I think I'd prefer SetSelection, especially given the other function Selection to get the selection offsets.

Yes, I very much expect the caret position to change when calling SetSelection (and vice versa). I don't want the user to have to worry about the caret and selection coming out of sync. That's the reason behind my "MoveAbs" suggestion, and the reason for my wanting to merge the selection positions with the caret position.

Reply:
Looking at this some more, I see that SetSelection, as it stands, breaks the usual invariant that the caret is at one side or the other of the selection, and that it should probably set the editor caret, too.  But to which side?  I guess that's your point: It should be renamed to MoveAbs, accept an offset and a signed length, move the caret to the offset, and set the selection start & end based on the length.

This might be a good place for another editor option to Move.  By default, Move should move relative the current caret, and update the selection along with it, just like it does now.  But I could add a MoveAbsolute flag, where "distance" is then interpreted as an absolute offset, and also a Selection length, which would update the selection.

That would also let you dispense with SetSelection, which would now be something more like `Move(0, WithSelection(<length>))`.

Reply:
(Looks like we were commenting at the same time.)